src/unix/fsevents.c

```diff
@@ -649,9 +649,15 @@ static int uv__fsevents_loop_init(uv_loop_t* loop) {
   if (pthread_attr_init(attr))
     attr = NULL;
 
+  /**
+   * macOS 10.15 FSEvents uses alloca() stack allocations, requiring a larger
+   * stack size. Leave this as the default 512KB size in that case.
+   */
+  #if MAC_OS_X_VERSION_MAX_ALLOWED < 101500
   if (attr != NULL)
     if (pthread_attr_setstacksize(attr, 4 * PTHREAD_STACK_MIN))
```

Comment:
I briefly considered bumping this value to `8` or something higher, but think we should remove the `pthread_attr_setstacksize` call entirely. Rationale:

1. This could break again in the future. We have no control over what Apple does with the stack in their kernel code. They likely assume that stack sizes below 512KB are safe since that's the default they set.
2. The `fsevents` npm module just passes `NULL` to `pthread_create`, so they're also using the default value: https://github.com/fsevents/fsevents/blob/fab136ace9f7f7a7a7856c2d684956aeac938856/src/fsevents.c#L81
3. macOS isn't an embedded environment where memory constraints are very low.
4. We only create 1 pthread for fsevents, so this isn't a N*(512-8)KB change. It's just a 504KB memory increase for users of `fs.watch`.

Reply:
Hitting the `Resolve conversation` button. We ended up bumping this even further to 8MB.